clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -204,7 +258,7 @@ public synchronized TransactionalRequestResult initializeTransactions() {
         ensureTransactional();
         transitionTo(State.INITIALIZING);
         setProducerIdAndEpoch(ProducerIdAndEpoch.NONE);
-        this.nextSequence.clear();
+        topicPartitionBookkeeper.reset();
```

Comment:
I think this call might not strictly be needed. We cannot return to the INITIALIZING state after beginning any transactions.